Replacement guide javadocs for deprecated classes, part one

elmot opened the pull request:
Partial fix for #8682

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/8709)
<!-- Reviewable:end -->

Commit:
Replacement guide javadocs for deprecated classes, part one

Partial fix for #8682

hesara commented:
Reviewed 52 of 102 files at r1.
Review status: 52 of 102 files reviewed at latest revision, 10 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Property.java, line 53 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7Uohu7FSzrltxEpxw:-Ke7Uohu7FSzrltxEpxx:b7l9qb0) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/Property.java#L53)):*
> ```Java
>  *
>  * @author Vaadin Ltd
>  * @deprecated As of 8.0, replaced by {@link ValueProvider}, {@link Setter}, see {@link Binder}
> ```

(perhaps could also mention HasValue - depends on how Property was used)

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java, line 56 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7VAufME6CuK3-BvO2:-Ke7VAufME6CuK3-BvO3:b-nlu06i) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java#L56)):*
> ```Java
>  * @author Vaadin Ltd.
>  * @since 3.0
>  * @deprecated As of 8.0, replaced by {@link Binder}
> ```

could perhaps mention the relevant method(s) in Binder, and that validation is no longer done by individual components but on the binding level - and as there is a new Validator interface could point to it, but note also that Binder also supports simpler validation

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/fieldgroup/BeanFieldGroup.java, line 32 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7VgBPYWImlcOIfeXI:-Ke7VgBPYWImlcOIfeXJ:b-gkmcv6) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/fieldgroup/BeanFieldGroup.java#L32)):*
> ```Java
> /**
>  *
>  * @deprecated As of 8.0, replaced by {@link Binder}
> ```

could also mention BeanValidationBinder

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java, line 51 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7WGwjKrTkOZCnQMTO:-Ke7WGwkbXwz1p9ggSo1:bqq5vyx) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java#L51)):*
> ```Java
>  * @since 3.0
>  *
>  * @deprecated As of 8.0, replaced by {@link DataProvider}
> ```

not really - this relates to hierarchy support planned for 8.1

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/sqlcontainer/SQLContainer.java, line 50 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7XLzbcmBUBCsprKlC:-Ke7XLzbcmBUBCsprKlD:b-ahnjt5) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/util/sqlcontainer/SQLContainer.java#L50)):*
> ```Java
> 
> /**
>  * @deprecated As of 8.0, no replacement available.
> ```

could refer to using a DataProvider

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/BaseTheme.java, line 35 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7YNrMy7DzIdj9THH4:-Ke7YNrMy7DzIdj9THH5:b8xjmoc) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/BaseTheme.java#L35)):*
> ```Java
>  *
>  * @since 6.3.0
>  * @deprecated As of 8.0, no replacement available.
> ```

could mention that the currently supported theme is Valo - that is what helps people if they just stumble upon a deprecation warning about an old theme

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/ChameleonTheme.java, line 19 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7YdTwJOlrQBye8iv_:-Ke7YdTwJOlrQBye8iva:b7fh0py) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/ChameleonTheme.java#L19)):*
> ```Java
> 
> /**
>  * @deprecated As of 8.0, no replacement available.
> ```

see BaseTheme

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Reindeer.java, line 26 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7Yg4LuD1WmDOyfVbI:-Ke7Yg4LuD1WmDOyfVbJ:b7fh0py) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Reindeer.java#L26)):*
> ```Java
> 
> /**
>  * @deprecated As of 8.0, no replacement available.
> ```

see BaseTheme

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Runo.java, line 19 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7YiGFzw-E2pNpnP7J:-Ke7YiGFzw-E2pNpnP7K:b7fh0py) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Runo.java#L19)):*
> ```Java
> 
> /**
>  * @deprecated As of 8.0, no replacement available.
> ```

see BaseTheme

---

*[compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java, line 29 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7Ymy_89pUvFEg5n1G:-Ke7Ymy_89pUvFEg5n1H:bjclreu) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java#L29)):*
> ```Java
>  * @author Vaadin Ltd.
>  * @since 7.0
>  * @see DateResolution
> ```

should also this be deprecation with a mention that the new components (DateField and DateTimeField) use DateResolution and DateTimeResolution, respectively

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

Commit:
Fixes after a review

elmot commented:
Review status: 48 of 102 files reviewed at latest revision, 10 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Property.java, line 53 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7Uohu7FSzrltxEpxw:-Ke7ozBk3r1KVDm3F6XJ:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/Property.java#L53)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(perhaps could also mention HasValue - depends on how Property was used)
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java, line 56 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7VAufME6CuK3-BvO2:-Ke7qRH7A2SI4RHrCiOL:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java#L56)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

could perhaps mention the relevant method(s) in Binder, and that validation is no longer done by individual components but on the binding level - and as there is a new Validator interface could point to it, but note also that Binder also supports simpler validation
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/fieldgroup/BeanFieldGroup.java, line 32 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7VgBPYWImlcOIfeXI:-Ke7qbo-VnKQmenlGOwq:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/fieldgroup/BeanFieldGroup.java#L32)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

could also mention BeanValidationBinder
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java, line 51 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7WGwjKrTkOZCnQMTO:-Ke7s8sN8NmTRMo5fD6h:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java#L51)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

not really - this relates to hierarchy support planned for 8.1
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/sqlcontainer/SQLContainer.java, line 50 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7XLzbcmBUBCsprKlC:-Ke7sIg8h8jl0gx-SfdM:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/util/sqlcontainer/SQLContainer.java#L50)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

could refer to using a DataProvider
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/BaseTheme.java, line 35 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7YNrMy7DzIdj9THH4:-Ke7snRJMlbISmwICu83:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/BaseTheme.java#L35)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

could mention that the currently supported theme is Valo - that is what helps people if they just stumble upon a deprecation warning about an old theme
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/ChameleonTheme.java, line 19 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7YdTwJOlrQBye8iv_:-Ke7spU1RDuG_NJIMCFZ:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/ChameleonTheme.java#L19)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

see BaseTheme
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Reindeer.java, line 26 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7Yg4LuD1WmDOyfVbI:-Ke7sqkDCZA8Kks-fFeE:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Reindeer.java#L26)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

see BaseTheme
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Runo.java, line 19 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7YiGFzw-E2pNpnP7J:-Ke7srkxhiy4IcG2rfPQ:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/ui/themes/Runo.java#L19)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

see BaseTheme
</blockquote></details>

Done.

---

*[compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java, line 29 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7Ymy_89pUvFEg5n1G:-Ke8KoTDt6lIF1kSW955:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java#L29)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

should also this be deprecation with a mention that the new components (DateField and DateTimeField) use DateResolution and DateTimeResolution, respectively
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

elmot commented:
Review status: 48 of 102 files reviewed at latest revision, 10 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java, line 51 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7WGwjKrTkOZCnQMTO:-Ke8M6v2ZqDN38yczbLX:b-76jtmf) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java#L51)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Added a comment to update this deprecation to the hierarchical support ticket. 

---

*[compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java, line 31 at r2](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke8Lq91jylf26PbaWLh:-Ke8Lq92G10xd05-Aw9A:boj7u45) ([raw file](https://github.com/vaadin/framework/blob/c10b81cdcd7c83a015eb53305129b9a6893d7bb9/compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java#L31)):*
> ```Java
>  *
>  * @deprecated As of 8.0, replaced by {@link DateResolution} and {@link DateTimeResolution},
>  * used by new {@code com.vaadin.ui.DateField} and {@code com.vaadin.ui.DateTimeField}
> ```

I can't use normal @link here because this is `vaadin-shared`and those classes are not visible.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 9 of 11 files at r2.
Review status: 57 of 102 files reviewed at latest revision, 2 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java, line 56 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7VAufME6CuK3-BvO2:-KeIVp6Ocmt22K26a6zX:ba6agh0) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java#L56)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Too many commas make this hard to read - using full stops and possibly other reformulations could help a little.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

Commit:
Too long sentence is rewritten

elmot commented:
Review status: 57 of 102 files reviewed at latest revision, 2 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java, line 56 at r1](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke7VAufME6CuK3-BvO2:-KeIo9V4qrW1Lg5XJo1k:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/71af7911b257c61a01d6900f495af349fc417a19/compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java#L56)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Too many commas make this hard to read - using full stops and possibly other reformulations could help a little.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

Commit:
Extra imports removed

Commit:
Some more clarifications are added.

hesara commented:
Reviewed 2 of 102 files at r1, 26 of 50 files at r3.
Review status: 60 of 102 files reviewed at latest revision, 6 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Container.java, line 667 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-Kf6Mnywn4N4HWEq7-zJ:-Kf6MnyxJg8747N-iKqk:bfugb1l) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Container.java#L667)):*
> ```Java
>      * </ul>
>      *
>      * @deprecated As of 8.0, no replacement available, planned to be replaced in 8.1.
> ```

(need to update this in a separate PR in master, but not pick that change to 8.0)

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validatable.java, line 21 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-Kf6N8oL3d3IYFICU9EC:-Kf6N8oL3d3IYFICU9ED:bfcr0w7) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Validatable.java#L21)):*
> ```Java
> import com.vaadin.data.Binder;
> import com.vaadin.data.ErrorMessageProvider;
> import com.vaadin.server.SerializablePredicate;
> ```

where did these imports come from?

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java, line 56 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGC_1LtbAYQJoWYQ37:-KfGC_1LtbAYQJoWYQ38:brgvb31) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java#L56)):*
> ```Java
>  * @since 3.0
>  * @deprecated As of 8.0, replaced by {@link com.vaadin.data.Validator}. The validation is performed
>  * outside components, see new {@link Binder} and its methods of name {@code withValidator(...)}
> ```

maybe just {@link Binder}.{@code withValidator(...)}

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/FilesystemContainer.java, line 46 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfBhIuvsfTdch8_X3nG:-KfBhIuwnMnoMe_tu5oH:b-gpncto) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/FilesystemContainer.java#L46)):*
> ```Java
>  * @since 3.0
>  *
>  * @deprecated As of 8.0, replaced by {@link DataProvider}
> ```

(not really, as hierarchy is not supported)

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/HierarchicalContainer.java, line 39 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGDLHuUNfoXPZz43Jl:-KfGDLHuUNfoXPZz43Jm:b-9wsi1t) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/HierarchicalContainer.java#L39)):*
> ```Java
>  * @since 3.0
>  *
>  * @deprecated As of 8.0, no replacement available.
> ```

mention 8.1?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 1 of 102 files at r1.
Review status: 61 of 102 files reviewed at latest revision, 6 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Collapsible.java, line 39 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGDz031BYG5Ay1oNKa:-KfGDz031BYG5Ay1oNKb:b-w3f3z3) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Collapsible.java#L39)):*
> ```Java
>  * <p>
>  *
>  * @deprecated No longer supported with Vaadin 8.0
> ```

hierarchy support coming in 8.1?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

hesara commented:
Review status: 61 of 102 files reviewed at latest revision, 7 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java, line 50 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGELhyAFd9HHM3TxxO:-KfGELhyAFd9HHM3TxxP:bl8kqg7) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java#L50)):*
> ```Java
>  * @since 3.0
>  *
>  * @deprecated As of 8.0, no replacement available, planned to be replaced in Vaadin 8.1.
> ```

maybe can be a bit more vague about when we'll provide the new hierarchical data support (here and elsewhere)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 5 of 102 files at r1, 15 of 50 files at r3.
Review status: 81 of 102 files reviewed at latest revision, 13 unresolved discussions.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Collapsible.java, line 39 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGDz031BYG5Ay1oNKa:-KfGF9SzZjspNqIW3GE1:b-g5n5iy) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Collapsible.java#L39)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

hierarchy support coming in 8.1?
</blockquote></details>

or more vague expression about the future

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/DefaultItemSorter.java, line 41 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGFnTPKNRuc68JRV8t:-KfGFnTPKNRuc68JRV8u:b-z8srq0) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/DefaultItemSorter.java#L41)):*
> ```Java
>  *
>  *
>  * @deprecated As of 8.0, no replacement available, see {@see AbstractBackEndDataProvider}
> ```

.setSortOrders() and Query.getSortOrders()

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/ItemSorter.java, line 29 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGFwFgLCF01D1qILbC:-KfGFwFhyORY7-gDGJ_4:bnznwjw) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/ItemSorter.java#L29)):*
> ```Java
>  * implementations to provide a custom sorting method.
>  *
>  * @deprecated As of 8.0, no replacement available.
> ```

see DefaultItemSorter comment

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertysetItem.java, line 38 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGGBRbcxQ5SREThjqP:-KfGGBRbcxQ5SREThjqQ:b-pd81zq) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertysetItem.java#L38)):*
> ```Java
>  * @since 3.0
>  *
>  * @deprecated As of 8.0, no replacement available.
> ```

can use a Map directly with Binder or whatever

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java, line 34 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGGI6h-hMaTvX5bILA:-KfGGI6h-hMaTvX5bILB:b-9n5ejg) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java#L34)):*
> ```Java
>  * @author Vaadin Ltd
>  *
>  * @deprecated As of 8.0, no replacement available.
> ```

can use lambdas as ValueProvider etc.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/filter/AbstractJunctionFilter.java, line 32 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGGYUIEved_bglFABR:-KfGGYUIEved_bglFABS:b-ittys0) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/filter/AbstractJunctionFilter.java#L32)):*
> ```Java
>  * @since 6.6
>  *
>  * @deprecated As of 8.0, no replacement available.
> ```

For in-memory DPs, use predicates as filters. For back-end DPs, filters are specific to the DP (implementation dependent). Same applies to other filters.

---

*[compatibility-server/src/main/java/com/vaadin/v7/event/ItemClickEvent.java, line 37 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGEkv7NquqbtOB94P-:-KfGEkv8ATYY-i7gBv6f:b-2l3w77) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/event/ItemClickEvent.java#L37)):*
> ```Java
>  * @since 5.3
>  *
>  * @deprecated As of 8.0, no replacement available, see component-specific click events.
> ```

remove the first part about no replacement

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

Commit:
Some updates after a discussion

hesara commented:
Reviewed 4 of 50 files at r3, 24 of 26 files at r4.
Review status: 102 of 104 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfHQyL7cZdQxKk7ZeO7:-KfHQyL7cZdQxKk7ZeO8:b-f06sra) (no related file):*
some comments + needs rebase/merge

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/BeanItem.java, line 42 at r4](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfHQl8PguGHXpe1syBL:-KfHQl8Qtb2CPGv2TXuS:b-xtxqdf) ([raw file](https://github.com/vaadin/framework/blob/6c55ed52739b44ac22ef7b481caf8214e280a7df/compatibility-server/src/main/java/com/vaadin/v7/data/util/BeanItem.java#L42)):*
> ```Java
>  * @deprecated As of 8.0, no direct replacement available. You can use any bean directly as an item for {@link Binder}
>  * or {@link DataProvider} and access the item properties with lambdas like {@code binder.forField(component).bind(...)} or
>  * {@code new Grid<Bean>(dataProvider).addColumn(bean->map.getSomething())}.
> ```

bean.getSomething() ?

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java, line 39 at r4](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfHQuSbNGWlk7Y5NDqZ:-KfHQuSbNGWlk7Y5NDq_:bcj3066) ([raw file](https://github.com/vaadin/framework/blob/6c55ed52739b44ac22ef7b481caf8214e280a7df/compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java#L39)):*
> ```Java
>  * @deprecated As of 8.0, no direct replacement available. You can use any bean directly as an item for {@link Binder}
>  * or {@link DataProvider} and access item properties with lambdas like {@code binder.forField(component).bind(...)} or
>  * {@code new Grid<Bean>(dataProvider).addColumn(bean->map.getSomething())}.
> ```

bean.getSomething()

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

Commit:
Typos fixed

elmot commented:
Review status: 101 of 103 files reviewed at latest revision, 17 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfHQyL7cZdQxKk7ZeO7:-KfHYri3f9DoN0EHD-08:bb6un1j) (no related file):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

some comments + needs rebase/merge
</blockquote></details>

Done

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Collapsible.java, line 39 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGDz031BYG5Ay1oNKa:-KfHHfbC8tQDtHPtJlI_:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Collapsible.java#L39)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

or more vague expression about the future
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Container.java, line 667 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-Kf6Mnywn4N4HWEq7-zJ:-KfHHnyjRUZw2FTz5f04:b-7yv2fe) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Container.java#L667)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(need to update this in a separate PR in master, but not pick that change to 8.0)
</blockquote></details>

What?

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validatable.java, line 21 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-Kf6N8oL3d3IYFICU9EC:-KfGuqWcYDH8pvmBNwu_:b-r7x8pj) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Validatable.java#L21)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

where did these imports come from?
</blockquote></details>

Done, removed.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java, line 56 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGC_1LtbAYQJoWYQ37:-KfGv5sJZJvyrrKpBeKp:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/Validator.java#L56)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

maybe just {@link Binder}.{@code withValidator(...)}
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/BeanItem.java, line 42 at r4](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfHQl8PguGHXpe1syBL:-KfHVSmJOMdFWxTCzKdj:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/6c55ed52739b44ac22ef7b481caf8214e280a7df/compatibility-server/src/main/java/com/vaadin/v7/data/util/BeanItem.java#L42)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

bean.getSomething() ?
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java, line 50 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGELhyAFd9HHM3TxxO:-KfH63FlWsz67Mlh-2v2:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/ContainerHierarchicalWrapper.java#L50)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

maybe can be a bit more vague about when we'll provide the new hierarchical data support (here and elsewhere)
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/DefaultItemSorter.java, line 41 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGFnTPKNRuc68JRV8t:-KfH65mSvQTE8KFdWSnq:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/DefaultItemSorter.java#L41)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

.setSortOrders() and Query.getSortOrders()
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/FilesystemContainer.java, line 46 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfBhIuvsfTdch8_X3nG:-KfH69JPicWULG-sS1r-:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/FilesystemContainer.java#L46)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(not really, as hierarchy is not supported)
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/HierarchicalContainer.java, line 39 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGDLHuUNfoXPZz43Jl:-KfH6BHQl8ews_k7DyRC:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/HierarchicalContainer.java#L39)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

mention 8.1?
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/ItemSorter.java, line 29 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGFwFgLCF01D1qILbC:-KfH0xo_7zVI43q5A4Dj:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/ItemSorter.java#L29)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

see DefaultItemSorter comment
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertysetItem.java, line 38 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGGBRbcxQ5SREThjqP:-KfHH_0Asf3fP_Ax7pfX:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertysetItem.java#L38)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

can use a Map directly with Binder or whatever
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java, line 34 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGGI6h-hMaTvX5bILA:-KfHHaCw4DC9sMEOfadt:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java#L34)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

can use lambdas as ValueProvider etc.
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java, line 39 at r4](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfHQuSbNGWlk7Y5NDqZ:-KfHV_Z9KbeaeR37IBxj:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/6c55ed52739b44ac22ef7b481caf8214e280a7df/compatibility-server/src/main/java/com/vaadin/v7/data/util/PropertyValueGenerator.java#L39)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

bean.getSomething()
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/data/util/filter/AbstractJunctionFilter.java, line 32 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGGYUIEved_bglFABR:-KfH10mSGWZ9VpOAVABG:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/data/util/filter/AbstractJunctionFilter.java#L32)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

For in-memory DPs, use predicates as filters. For back-end DPs, filters are specific to the DP (implementation dependent). Same applies to other filters.
</blockquote></details>

Done.

---

*[compatibility-server/src/main/java/com/vaadin/v7/event/ItemClickEvent.java, line 37 at r3](https://reviewable.io:443/reviews/vaadin/framework/8709#-KfGEkv7NquqbtOB94P-:-KfGRPTPYDm7OnTK8lzP:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8ca15fbff3be92065e85c375aebb736c45691a59/compatibility-server/src/main/java/com/vaadin/v7/event/ItemClickEvent.java#L37)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

remove the first part about no replacement
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 2 of 3 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->

hesara commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709#-:-KfHZ_Kxb2Wx2sFL5xro:bnfp4nl)*
<!-- Sent from Reviewable.io -->

hesara commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java, line 31 at r2](https://reviewable.io:443/reviews/vaadin/framework/8709#-Ke8Lq91jylf26PbaWLh:-KfHZhIO8gyVkG4VG-yh:b32wtd) ([raw file](https://github.com/vaadin/framework/blob/c10b81cdcd7c83a015eb53305129b9a6893d7bb9/compatibility-shared/src/main/java/com/vaadin/v7/shared/ui/datefield/Resolution.java#L31)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

I can't use normal @link here because this is `vaadin-shared`and those classes are not visible.
</blockquote></details>

ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8709)*
<!-- Sent from Reviewable.io -->
